analyzers/src/SonarAnalyzer.CSharp/Rules/ReturnEmptyCollectionInsteadOfNull.cs

```diff
@@ -23,131 +23,119 @@ namespace SonarAnalyzer.Rules.CSharp
-                KnownType.System_Array
-            );
+        private static readonly ImmutableArray<KnownType> CollectionTypes = ImmutableArray.Create(
+            KnownType.System_Collections_IEnumerable,
+            KnownType.System_Array);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
-            context.RegisterSyntaxNodeActionInNonGenerated(ReportIfReturnsNull,
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(ReportIfReturnsNullOrDefault,
                 SyntaxKind.MethodDeclaration,
                 SyntaxKindEx.LocalFunctionStatement,
-                SyntaxKind.PropertyDeclaration);
-        }
+                SyntaxKind.PropertyDeclaration,
+                SyntaxKind.OperatorDeclaration);
 
-        private static void ReportIfReturnsNull(SyntaxNodeAnalysisContext context)
+        private static void ReportIfReturnsNullOrDefault(SyntaxNodeAnalysisContext context)
         {
             var expressionBody = GetExpressionBody(context.Node);
-            if (expressionBody != null)
+            if (expressionBody is not null)
             {
-                var arrowedNullLiteral = GetNullLiteralOrDefault(expressionBody);
-                if (arrowedNullLiteral != null &&
-                    IsReturningCollection(context))
+                var nullOrDefaultLiterals = GetNullOrDefaultExpressions(expressionBody.Expression)
+                    .Select(statement => statement.GetLocation())
+                    .ToList();
+
+                if (nullOrDefaultLiterals.Count > 0 && IsReturningCollection(context))
                 {
-                    context.ReportIssue(Diagnostic.Create(rule, arrowedNullLiteral.GetLocation()));
+                    context.ReportIssue(Rule.CreateDiagnostic(context.Compilation, nullOrDefaultLiterals[0], additionalLocations: nullOrDefaultLiterals.Skip(1)));
                 }
 
                 return;
             }
 
             var body = GetBody(context.Node);
-            if (body != null)
+            if (body is not null)
             {
-                var returnNullStatements = GetReturnNullStatements(body)
+                var nullOrDefaultLiterals = GetReturnNullOrDefaultExpressions(body)
                     .Select(returnStatement => returnStatement.GetLocation())
                     .ToList();
-                if (returnNullStatements.Count > 0 &&
-                    IsReturningCollection(context))
+
+                if (nullOrDefaultLiterals.Count > 0 && IsReturningCollection(context))
                 {
-                    context.ReportIssue(rule.CreateDiagnostic(context.Compilation, returnNullStatements[0], additionalLocations: returnNullStatements.Skip(1)));
+                    context.ReportIssue(Rule.CreateDiagnostic(context.Compilation, nullOrDefaultLiterals[0], additionalLocations: nullOrDefaultLiterals.Skip(1)));
                 }
```

Comment:
This if block is repeated in both cases with body and expression body. I suggest extracting it to its own method and reducing duplication.